nidmviewer/viewer.py

```diff
@@ -92,6 +94,16 @@ def generate(ttl_files,base_image=None,retrieve=False,view_in_browser=False,colu
 
     if view_in_browser==True:
         peaks,copy_list = generate_temp(peaks,"excsetmap_location")
+        # Check if the nifti is empty
+        for exc_set_file in copy_list.keys():
+            img = nib.load(exc_set_file)
+            data = img.get_data()
+            data = np.nan_to_num(data)
+            if np.count_nonzero(data) == 0:
+                # Empty excursion set is removed from the display list
+                del copy_list[exc_set_file]
+                peaks[peaks.keys()[0]] = {}
+
         if base_image == None:
```

Comment:
Did you walk through this function? It doesn't seem to work - there are many peaks in my image that are definitely empty:

![image](https://cloud.githubusercontent.com/assets/814322/15652355/73e3aff6-263a-11e6-95a8-76142d5c6547.png)

Please tell me if I have this correct - I had thought that the function should only show peaks on the image that don't have a value of 0 in the data. The function you modified above is at the step when the final thing is rendered into a browser - the copy list is the set of images referenced in peaks - and it looks like you delete the entire image only if the entire data is empty (would this ever be the case)? I had thought we would want to prune away peak coordinates that are empty, vs. the whole image? @chrisfilo is this the desired functionality?